### PR TITLE
Allow html in titles and correctly store all title overrides from single HTML

### LINKED ISSUE #77: Titles in nav are not being used for translucent binders when adapting from single HTML
All pages and chapters / units directly under the book have their title overrides stored, but everything in chapters / units have no title overrides.

Part of https://github.com/Connexions/cnx-recipes/issues/34

### PR DESCRIPTION
**Pull request by karenc**
Close #77 and #78.

### CODE REVIEW

File: cnxepub/formatters.py [hunk 1]
```diff
@@ -571,12 +571,14 @@ def html_listify(tree, root_xl_element, extensions, list_type='ol'):
     for node in tree:
         li_elm = etree.SubElement(root_xl_element, 'li')
         if node['id'] == 'subcol':
-            span_elm = etree.SubElement(li_elm, 'span')
-            span_elm.text = node['title']
+            span_elm = lxml.html.fragment_fromstring(
+                node['title'], create_parent='span')
```
Comment (reedstrm) [inline cnxepub/formatters.py]: Won't this _force_ all titles to be HTML, and adds an extra wrapping <span> tag around all them, to boot?

File: cnxepub/html_parsers.py [hunk 2]
```diff
@@ -16,11 +18,13 @@
...
 
-def _squash_to_text(elm):
+def _squash_to_text(elm, remove_namespaces=False):
     value = [elm.text or '']
     for child in elm.getchildren():
-        value.append(etree.tostring(child).decode('utf-8'))
+        value.append(etree.tostring(child).decode('utf-8').strip())
         value.append(child.tail or '')
+    if remove_namespaces:
+        value = [re.sub(' xmlns:?[^=]*="[^"]*"', '', v) for v in value]
```
Comment (reedstrm) [inline cnxepub/html_parsers.py]: sniff sniff - processing structured data w/ regexs usually means something is very wrong ... This also only allows for one level of markup in titles.

File: cnxepub/formatters.py (hunk 1 above)
Comment (karenc) [inline cnxepub/formatters.py]: No, `span_elm = etree.SubElement(li_elm, 'span')`, we've always had a `<span>` around the title.

File: cnxepub/html_parsers.py (hunk 2 above)
Comment (karenc) [inline cnxepub/html_parsers.py]: I don't see a way to remove namespaces.  If you don't like it I guess we don't have to remove the namespaces.

Why do you think this only allows one level of markup?

File: cnxepub/formatters.py (hunk 1 above)
Comment (reedstrm) [inline cnxepub/formatters.py]: Ok. yes I see that now. Hmm. We may need to change that if Helene et al object, but it's the minimal change for now, agreed.

File: cnxepub/html_parsers.py (hunk 2 above)
Comment (reedstrm) [inline cnxepub/html_parsers.py]: Ah, you're right it gets them all - I wasn't seeing recursion, but etree.tostring() does the entire subtree.

Comment (reedstrm) [inline cnxepub/html_parsers.py]: The problem with removing the namespace declarations like this is that we don't also remove the use of the namespace, we may end up with orphan tags. Admittedly, for our use-case the namespaces are almost always in the way - see my PR re: extending a bunch of xpaths to match both namespaced and non-namespaced (xhtml + html). The only trick I know for stripping them in a structured way is to create a root node with the default namespace set to what you want to remove, parse w/ ns_clean=True, then ... hehe I think it ends up using string manipulation in the end, to remove the default xmlns declaration. Hmm.

**Pull request merged**